cmd/kube-controller-manager/app/options/options_test.go

```diff
@@ -171,8 +169,7 @@ func TestAddFlags(t *testing.T) {
 	expected := &KubeControllerManagerOptions{
 		Generic: &cmoptions.GenericControllerManagerConfigurationOptions{
 			GenericControllerManagerConfiguration: &cmconfig.GenericControllerManagerConfiguration{
-				Port:            10252,     // Note: InsecureServingOptions.ApplyTo will write the flag value back into the component config
-				Address:         "0.0.0.0", // Note: InsecureServingOptions.ApplyTo will write the flag value back into the component config
+				Address:         "0.0.0.0", // Note: This field should have no effect in CM now, and "0.0.0.0" is the default value.
```

Comment:
@sttts - I expected these values to be defaulted and effective. If this has no effect, what field controls the listen address/port?

Reply:
IIUC these settings seem to be used for the insecure serving, I'd like @sttts to confirm.

> If this has no effect, what field controls the listen address/port?
Flag `--bind-address` and `--secure-port`:
* https://github.com/kubernetes/kubernetes/blob/75463bda1d9103a46de127528ee7b3b96ceada54/cmd/kube-controller-manager/app/options/options.go#L235
* https://github.com/kubernetes/kubernetes/blob/75463bda1d9103a46de127528ee7b3b96ceada54/cmd/kube-controller-manager/app/options/options.go#L357-L367
* https://github.com/kubernetes/kubernetes/blob/75463bda1d9103a46de127528ee7b3b96ceada54/cmd/kube-controller-manager/app/controllermanager.go#L193-L200

Reply:
if port and address are unused, should they be removed? I'm surprised the insecure serving config wasn't represented in the config types

Reply:
Yes, would like to see them removed from `GenericControllerManagerConfiguration`. Today they are part of the v1alpha1 config type. Is that used anywhere? Compare k8s.io/controller-manager/config/v1alpha1/types.go